source/model/backtraceMetrics.ts

```diff
@@ -0,0 +1,228 @@
+
+  /**
+   * Send POST to summed-events API endpoint
+   */
+  private async sendSummedEvent(
+    guid: string,
+    sessionId: string,
+    metricGroup: string,
+  ): Promise<void> {
+    const endpoint = `${this.hostname}/api/summed-events/submit?universe=${this.universe}&token=${this.token}`;
+
+    const payload = {
+      application: this.applicationName,
+      appversion: this.applicationVersion,
+      metadata: {
+        dropped_events: 0,
+      },
+      summed_events: [
+        {
+          timestamp: this.timestamp,
+          metric_group: metricGroup,
+          attributes: {
+            guid,
+            'uname.sysname': this.userAgent,
+            'application.version': this.applicationVersion,
+            'application.session': sessionId, // application session ?
+          },
+        },
+      ],
+    };
+
+    await this.post(endpoint, JSON.stringify(payload));
+    return;
+  }
+
+  /**
+   * Send POST request.
+   * @param url - string endpoint
+   * @param data - stringified JSON object
+   */
+  private async post(url: string, data: string): Promise<void> {
+    try {
+      return new Promise<void>((res, rej) => {
+        const http = new XMLHttpRequest();
+        http.timeout = this.timeout;
+        http.open('POST', url, true);
+        http.setRequestHeader('Content-type', 'application/json');
+        http.send(data);
+        http.onload = (e) => {
+          if (http.readyState !== XMLHttpRequest.DONE){
+            return
+          }
+
+          if (http.status === 200) {
+            res();
+          } else if (http.status === 429) {
+            rej('Backtrace - reached metric limit.');
+          } else {
+            rej(
+              `Invalid attempt to submit metric to Backtrace. HTTP Error ${http.status}. Result: ${http.responseText}`,
+            );
+          }
+        };
+        http.onerror = (e) => {
+          rej(e);
+        };
+      });
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  /**
+   * Get stored sessionId
+   */
+  private getSessionId(): string | undefined {
+    return localStorage.getItem('sessionId') || undefined;
+  }
+
+  /**
+   * Create new sessionId and set local sessionId and sessionStart.
+   */
+  private createNewSession(): string {
+    const newSessionId = uuid();
+    localStorage.setItem('sessionId', newSessionId);
+    localStorage.setItem('sessionStart', this.timestamp.toString());
+    return newSessionId;
+  }
+
+  /**
+   * Get stored time since last page navigation or current time
+   */
+  private getSessionStart(): number | undefined {
+    const sessionStartStr = localStorage.getItem('sessionStart');
+    return sessionStartStr ? parseInt(sessionStartStr, 10) : undefined;
+  }
+
+  /**
+   * Get stored time since last session created or current time
+   */
+  private getLastActive(): number {
+    const lastActiveStr = localStorage.getItem('lastActive');
+    if (!lastActiveStr) {
+      this.setLastActive(this.timestamp);
```

Comment:
why getter setting something in the class definition? Should we set it somewhere else?

Reply:
Refactored this logic, no longer setting in this function